viewer/packages/camera-manager/src/Flexible/FlexibleControlsRotationHelper.ts

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import { Quaternion, Spherical, Vector3 } from 'three';
+import { FlexibleControls } from './FlexibleControls';
+import { FlexibleControlsType } from './FlexibleControlsType';
+
+const UP_VECTOR = new Vector3(0, 1, 0);
+const RIGHT_VECTOR = new Vector3(1, 0, 0);
+
+/**
+ * Helper class for rotating the camera. Call begin() before rotation,
+ * and end() after so the camera position updates correcly
+ * @beta
+ */
+export class FlexibleControlsRotationHelper {
+  private readonly oldOffset: Vector3 = new Vector3();
+  private readonly oldCameraVectorEnd: Spherical = new Spherical();
```

Comment:
If works for all types. OrbitInCenter you don't need to do anything in begin(). For First-person nothing at all, since the  camera vecor is changing only. This class hides away this complexity